### ISSUE TITLE
LuxSeries causing dtype to be erased for some attributes in the DataFrame

### ISSUE THREAD

**Issue by westernguy2** [lux-org/lux#140]
When running the following code: 

```python
df = pd.read_csv("../lux-datasets/data/airbnb_nyc.csv")
df
```
We get the following error 
``` 
ufunc 'isinf' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```
![Screen Shot 2020-11-15 at 4 00 31 PM](https://user-images.githubusercontent.com/32151899/99200696-b2311380-275b-11eb-9286-678a9d4a7b9e.png)

The error is raised because the dtype for the `name` column is set to None when running with `lux`. The dtype is set to None before `_repr_html_` is called, but rather it's set at `__init__` or when `df.dtypes` is requested.

The dtype is not None when `lux` is not being imported. It's also not None when the `LuxSeries` implementation is disabled (this was done by commenting out `_constructor_sliced`. 

In addition, the only function in `LuxSeries` to be called is `__init__`. Overall, it's unclear what's causing this issue. It doesn't seem to happen to other datasets, so it's possible this might be caused by a unique character in the `name` column of the dataset.

**Comment by dorisjlee** [comment]
The same issue is happening with the Spotify dataset
```python
df = pd.read_csv("../lux-datasets/data/spotify.csv")
df
```
![image](https://user-images.githubusercontent.com/5554675/100445094-7dbc4080-30e7-11eb-92aa-ea2c0bc86684.png)

**Comment by dorisjlee** [comment]
Coincidentally in both datasets, the dtype of the `name` column is what is being set to None. I suspect that the special keyword `name` property of Series was what was causing the issue. I tested this by renaming the column to a different name and the error went away.
<img src="https://user-images.githubusercontent.com/5554675/100517143-ab76b780-31c3-11eb-9d72-07162c2a47a7.png" width="35%"></img>

It seems like since the `name` property was not listed under the `_metadata` field, it was not carried over when we do:
```python
s.__dict__[attr] = getattr(self, attr, None)
```
inside the `_constructor_sliced`. So the problem was fixed when I added "name" to the metadata field. We should look into whether there are other properties native to `pd.DataFrame` or `pd.Series` that might cause issues (e.g., dtype, values; see `@property` in [generic.py](https://github.com/pandas-dev/pandas/blob/91abd0ababfca8b229e4b5ba64a26811c3da384a/pandas/core/generic.py#L5577))
